model.py

```diff
@@ -0,0 +1,60 @@
+def read_phonebook():
+    file = open('file.txt', 'r', encoding = 'utf-8')
+    lines = file.readlines()
+    book = []
+    for line in lines:
+        book.append(line)
+    file.close()
+    return book  
+
+
+
+
+def add_contact():
+    first_name = input("First name: ")
+    family_name = input("Family name: ")
+    phone_num = input("Phone number: ")
+    file = open('file.txt', 'a')
+    file.write(first_name + ' ')
+    file.write(family_name + ' ')
+    file.write(phone_num + '\n')
+    file.close()
+
+def find():
+    look_for = input("Enter a key word: ").lower()
+    lines = read_phonebook()
+    count = 0
+    for line in lines:
+        if look_for in line.lower():
+            count += 1
+            print(line)
+            break
+    else:
+        print("The record is not found")
+
+
+# Я не понимаю, почему функция не работает. Мне выдает ошибку "replace() argument 2 must be str, not None"
+def change_record():
+    to_change = input("Enter a name or number to change: ").lower()
+    lines = read_phonebook()
+    file = open('file.txt', 'w', encoding='utf-8')
+    for i in range(len(lines)):
+        if to_change in lines[i].lower():
+            lines[i] = lines[i].replace(to_change, add_contact())
+            file.write(lines[i])
```

Comment:
Наталья, ключ к разрешению проблем в правильном понимании ошибок. Прочтите внимательно. Для этого не нужно иметь высокий уровень знания английского. Тут говориться, что вторым аргументом метода `replace`, должен быть объект с типом данных `str`, но никак не `None`. Но функция add_contact у вас как раз `None` и возвращает.

Reply:
Тут вам не нужно использовать функцию `add_contact`, ведь она записывает контакт в файл, но первые три строки из этой функции вам пригодятся. Создайте новую функцию, например, с именем `ask` и пусть она возвращает строку, которую сформируют переменные `first_name`, `family_name` и `phone_num`.
```suggestion
            print(lines[i])  # чтобы посмотреть, что мы изменяем
            file.write(ask())
            break  # на случай, если будет несколько совпадений. Можно и убрать
```